src/Microsoft.OData.Client/IDataServiceContext.cs

```diff
@@ -0,0 +1,141 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.IO;
+using System.Linq.Expressions;
+using System.Net;
+using System.Threading.Tasks;
+
+namespace Microsoft.OData.Client
+{
+    public interface IDataServiceContext
+    {
+        DataServiceResponsePreference AddAndUpdateResponsePreference { get; set; }
+        bool ApplyingChanges { get; }
+        Uri BaseUri { get; set; }
+        DataServiceClientConfigurations Configurations { get; }
+        ICredentials Credentials { get; set; }
+        bool DisableInstanceAnnotationMaterialization { get; set; }
+        bool EnableWritingODataAnnotationWithoutPrefix { get; set; }
+        ReadOnlyCollection<EntityDescriptor> Entities { get; }
+        EntityParameterSendOption EntityParameterSendOption { get; set; }
+        EntityTracker EntityTracker { get; set; }
+        DataServiceClientFormat Format { get; }
+        bool IgnoreResourceNotFoundException { get; set; }
+        ReadOnlyCollection<LinkDescriptor> Links { get; }
+        ODataProtocolVersion MaxProtocolVersion { get; }
+        MergeOption MergeOption { get; set; }
+        Func<string, Uri> ResolveEntitySet { get; set; }
+        Func<Type, string> ResolveName { get; set; }
+        Func<string, Type> ResolveType { get; set; }
+        SaveChangesOptions SaveChangesDefaultOptions { get; set; }

```

Comment:
I don't like to add an interface and add all public members into the interface.

Reply:
Yes will restructure the interfaces.